templates/commands/render/action_include.go

```diff
@@ -126,3 +150,35 @@ func includePath(ctx context.Context, inc *spec.IncludePath, sp *stepParams) err
 	}
 	return nil
 }
+
+// TODO(#158): add support for ignoring file name. For example, "/foo" ignores
+// a single file named foo in the root, but "foo" ignores all files named foo
+// anywhere.
+// checkIgnore checks the given path against the given patterns, if given
+// patterns is not provided, a default list of patterns is used.
+func checkIgnore(patterns []model.String, path string) (bool, error) {
+	if len(patterns) > 0 {
```

Comment:
I see the motivation here for copy-pasting this code block, but I'm strongly opposed to copy-pasting code most of the time, because I've seen the problems that it creates in the long run. The different copies tend to diverge and accumulate separate bugs.

In this case, I think it's OK to just declare the the defaultIgnorePatterns as a list of model.String. There's already some special-case logic for handling the case where the position is undefined: https://github.com/abcxyz/abc/blob/c9d99ef4f1b4ea314235734b442463865a5132bb/templates/model/pos.go#L58, so it's kinda OK to have a model.String with an undefined position.

So, to summarize, my recommendation is:

Declare the defaultIgnorePatterns as `[]model.String`:
```
// defaultIgnorePatterns to be used if ignore is not provided.
var defaultIgnorePatterns = []model.String{
	{Val: ".DS_Store"},
	{Val: ".bin"},
	{Val: ".ssh"},
}
```

And simplify checkIgnore:
```
func checkIgnore(patterns []model.String, path string) (bool, error) {
	if len(patterns) == 0 {
		patterns = defaultIgnorePatterns
	}
	for _, p := range patterns {
		matched, err := filepath.Match(filepath.FromSlash(p.Val), path)
		if err != nil {
			return false,
				p.Pos.Errorf("failed to match path (%q) with pattern (%q): %w", path, p.Val, err)
		}
		if matched {
			return true, nil
		}
	}
	return false, nil
}
```

Reply:
Got it, was not sure about using nil position. looks much cleaner!